chai/source.py

```diff
@@ -0,0 +1,169 @@
+    - INSTANCE: https://github.com/tlaplus-community/tlaplus-standard/blob/3cbb0c251d63d5365f71aa0b3fea0719b6879edd/grammar/TLAPlus2Grammar.tla#L141-L144 # noqa
+
+    And for examples of both: http://lamport.azurewebsites.net/tla/newmodule.html#Section2
+    """
+    deps = []
+    # Comma separated deps may extend over many lines
+    in_comma_sep_deps = False
+    for line in module.splitlines():
+        if not in_comma_sep_deps and re.search(EXTENDS_LINE_PREFIX_RE, line):
+            in_comma_sep_deps = True
+
+        if not in_comma_sep_deps and re.search(INSTANCE_LINE_PREFIX_RE, line):
+            if "," in line:
+                in_comma_sep_deps = True
+            else:
+                new_deps = _get_dep_from_instance_line(line)
+                deps.extend(new_deps)
+
+        if in_comma_sep_deps:
+            new_deps = _get_comma_separated_deps(line)
+            # Check if we have a blank line,
+            # in which case we don't add anything and keep searching
+            if new_deps:
+                deps.extend(new_deps)
+                # If the line ends in a comma, we will have more deps to come
+                if not line.rstrip().endswith(","):
+                    in_comma_sep_deps = False
+
+    return deps
+
+
+def _load_deps_of_tla_file(tla_module: Path) -> List[str]:
+    """Load all the found dependencies on disk for TLA+ module
+
+    Return:
+        a list of the contents of the dependenceis
+    """
+    print(">>>", tla_module)
+    content = tla_module.read_text()
+    deps = _get_module_deps(content)
+    print(deps)
+    return [
+        dep
+        for f in tla_module.parent.resolve().iterdir()
+        if f.stem in deps and f.suffix.lstrip(".") == TLA_SUFFIX
+        for dep in (_load_deps_of_tla_file(f) + [f.read_text()])
+    ]
+
+
+class Source:
+    """
+    A source from which the client can load data
+    """
+
+    # Supported inputs to derive a `Source`
+    # Strings should be a
+    Input = Union[str, Path]
+
+    @classmethod
+    def load_input(cls, source: Input) -> str:
+        """Convert an Input into a string:
+
+        - loading the contents of a file specified by a `Path`
+        - acting as identity on a string
+        """
+        if isinstance(source, str):
+            return source
+        elif isinstance(source, Path):
+            return source.read_text()
+        else:
+            raise ValueError(
+                "Source can only be construced from a str or a Path,"
+                f"given {type(source)}"
+            )
+
+    @classmethod
+    def of_file(cls, p: Path, aux: List[Path]) -> Self:
+        """Create a Source for use in an RPC
+
+        Args:
+            p: The main file
+            aux: Any auxiliary files required as dependencies
+        """
+        return Source(
+            source=p.read_text(),
+            aux=[s.read_text() for s in aux],
+            format=p.suffix.lstrip("."),
```

Comment:
I'd add a comment here, for readability:

```suggestion
            format=p.suffix.lstrip("."),  # last filename suffix, without leading "."
```